### PR TITLE
ArangoSearch wildcard query

### PR DESCRIPTION
**Pull request by gnusi**
add support of `LIKE` function to ArangoSearch

### CODE REVIEW

**Comment by gnusi** [comment]
https://github.com/arangodb/backlog/issues/631

**Comment by gnusi** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7576/

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7577/

File: 3rdParty/iresearch/core/search/limited_sample_scorer-heap.hpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+               const sub_reader& reader,
+               const seek_term_iterator& term_itr);
+
+  void score(const index_reader& index,
+             const order::prepared& order,
+             std::vector<bstring>& stats);
+
+ private:
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief a representation of a term cookie with its asociated range_state
```
Comment (iurii-i-popov) [inline 3rdParty/iresearch/core/search/limited_sample_scorer-heap.hpp]: asSociated

File: 3rdParty/iresearch/core/search/limited_sample_scorer.cpp [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+    // find the stats for the current term
+    const auto res = map_utils::try_emplace(
+      term_stats,
+      make_hashed_ref(bytes_ref(scored_state.term), std::hash<bytes_ref>()),
+      index, field, order, stats_offset);
+
+    // find term attributes using cached state
+    if (!term_itr->seek(bytes_ref::NIL, *(scored_state.cookie))) {
+      continue; // some internal error that caused the term to disapear
```
Comment (iurii-i-popov) [inline 3rdParty/iresearch/core/search/limited_sample_scorer.cpp]: disapPear

File: 3rdParty/iresearch/core/search/limited_sample_scorer.hpp [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+               const sub_reader& reader,
+               const seek_term_iterator& term_itr);
+
+  void score(const index_reader& index,
+             const order::prepared& order,
+             std::vector<bstring>& stats);
+
+ private:
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief a representation of a term cookie with its asociated range_state
```
Comment (iurii-i-popov) [inline 3rdParty/iresearch/core/search/limited_sample_scorer.hpp]: asSociated

File: 3rdParty/iresearch/core/store/directory_attributes.cpp [hunk 4]
```diff
@@ -65,11 +65,11 @@ DEFINE_FACTORY_DEFAULT(fd_pool_size)
 
 const size_t FD_POOL_DEFAULT_SIZE = 8;
 
-fd_pool_size::fd_pool_size() NOEXCEPT
+fd_pool_size::fd_pool_size() noexcept
   : size(FD_POOL_DEFAULT_SIZE) { // arbitary size
```
Comment (iurii-i-popov) [inline 3rdParty/iresearch/core/store/directory_attributes.cpp]: arbitRary

File: 3rdParty/iresearch/core/search/limited_sample_scorer-heap.cpp [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+    // find the stats for the current term
+    const auto res = map_utils::try_emplace(
+      term_stats,
+      make_hashed_ref(bytes_ref(scored_state.term), std::hash<bytes_ref>()),
+      index, field, order, stats_offset);
+
+    // find term attributes using cached state
+    if (!term_itr->seek(bytes_ref::NIL, *(scored_state.cookie))) {
+      continue; // some internal error that caused the term to disapear
```
Comment (iurii-i-popov) [inline 3rdParty/iresearch/core/search/limited_sample_scorer-heap.cpp]: disapPear

File: 3rdParty/iresearch/core/formats/formats_burst_trie.cpp [hunk 6]
```diff
@@ -603,103 +751,402 @@ class term_iterator final : public irs::seek_term_iterator {
...
+    return true;
+  }
+
+  virtual void read() override {
+    assert(cur_block_);
+    term_iterator_base::read(*cur_block_);
+  }
+
+  virtual doc_iterator::ptr postings(const flags& features) const override {
+    return term_iterator_base::postings(cur_block_ ? cur_block_: nullptr, features);
```
Comment (iurii-i-popov) [inline 3rdParty/iresearch/core/formats/formats_burst_trie.cpp]: cur_block_, features

**Comment by gnusi** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7655/

**Pull request merged**